catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -452,60 +484,158 @@ private void moveToFailedIngestDirectory(File source) {
 
     private void buildQueue(Stream<Path> ingestStream, ArrayBlockingQueue<Metacard> metacardQueue,
             long start) {
-        ingestStream.filter(a -> !a.toFile()
-                .isDirectory())
-                .forEach(a -> {
-                    File file = a.toFile();
-
-                    if (file.isHidden()) {
-                        ignoreCount.incrementAndGet();
-                    } else {
-                        String extension = file.getName();
-
-                        if (extension.contains(".")) {
-                            int x = extension.indexOf('.');
-                            extension = extension.substring(x);
-                        }
 
-                        if (ignoreList != null && (ignoreList.contains(extension)
-                                || ignoreList.contains(file.getName()))) {
+        if (includeContent) {
+            File inputFile = new File(filePath);
+            Map<String, Serializable> arguments = new HashMap<>();
+            arguments.put(DumpCommand.FILE_PATH, inputFile.getParent() + File.separator);
+            arguments.put(FILE_NAME, inputFile.getName());
+
+            InputCollectionTransformer zipTransformer = getInputCollectionTransformer();
+            if (zipTransformer != null) {
+
+                ByteSource byteSource = com.google.common.io.Files.asByteSource(inputFile);
+
+                try (InputStream inputStream = byteSource.openBufferedStream()) {
+                    List<Metacard> metacardList = zipTransformer.transform(inputStream, arguments);
+                    if (metacardList.size() != 0) {
+                        metacardFileMapping = generateFileMap(inputFile.getParentFile());
+                        fileCount.set(metacardList.size());
+                        metacardQueue.addAll(metacardList);
+                    }
+                } catch (IOException | CatalogTransformerException e) {
+                    LOGGER.error("Unable to transform zip file into metacard list {}", e);
+                    INGEST_LOGGER.error("Unable to transform zip file into metacard list {}", e);
+                }
+            }
+        } else {
+
+            ingestStream.filter(a -> !a.toFile()
```

Comment:
Since it appears that only the `File` is used in the `forEach`, this would be more explicit:
```
ingestStream.map(Path::toFile)
    .filter(file -> !file.isDirectory())
    .forEach(file -> ...);
```